### PR TITLE
Reworked some documentation for clarity

### LINKED ISSUE #160: Documentation updates
**Describe the bug**
No bug. I'm just going through the docs as I learn about this project and would like to make modifications to increase the clarity.  I have a PR in the wings but wanted to file an issue first.

**To Reproduce**
n/a

**Expected behavior**
n/a

**Additional context**
Mostly these are just grammatical changes.  I've noted some opportunities to improve the clarity with some minor reworking of the structure, but nothing huge.  I also noted that there may be a bug in the pseudo code of the `RuleEngine_en.md` file that I've "fixed"  Basically, I think you need some extra `AND` elements in the conditions for the `< 1500` and `< 2000` rules because a fact where the price after tax is between `1000` and `1499.99` would match both and there's no priority associated with the rules.

### PR DESCRIPTION
**Pull request by derekwyatt**
* resolves #160
* Mostly just grammar and structural stuff
* I did make one more substantive change though, to the
  `RuleEngine_en.md` file.  I changed the rule pseudo code because I
  don't think the original was as deterministic as one might have
  expected. An item whose purchase price after tax is 1048USD (or
  anything between 1000USD and 1499.99USD) actually matches the < 1500
  and < 2000 rule in a single cycle.

### CODE REVIEW

File: docs/RETE_en.md [hunk 1]
```diff
@@ -132,33 +132,34 @@ Expression "Fact.B + Fact.Func(Fact.C)" --> A math operation contains 2 variable
 Expression "(Fact.B + Fact.Func(Fact.C))" - 20 -- A math operation also contains 2 variable.
 ```
 
-Each of the above Expressions will be remembered for their underlaying values whenever
-they get evaluated for the first time. So subsequent evaluation will not be evaluated
-as their remembered value will immediately returned.
+The resulting values for each of the above `Expression`s will be remembered (memoized) upon their first invocation so that subsequent references to them will avoid a re-invocation of them, returning the remembered value immediately instead.
```
Comment (newm4n) [inline docs/RETE_en.md:135]: memorized

Comment (newm4n) [reply docs/RETE_en.md:135]: Whoops, i don't know if `memoize` is a valid word. TIL.

**Review by newm4n — APPROVED** [review]
That's really a lot of good work there, we all, opensource community will surely can't thank you enough.  Thank you for you generous contribution. I will merge this now.

**Comment by derekwyatt** [comment]
You're most welcome!  Glad I could help.

**Pull request merged**